packages/strapi-admin/admin/src/containers/Webhooks/EditView/utils/formatData.js

```diff
@@ -0,0 +1,40 @@
+import { set } from 'lodash';
+
+const cleanData = data => {
+  const webhooks = data;
+  const headers = cleanHeaders(data.headers);
+
+  set(webhooks, 'headers', unformatHeaders(headers));
+  return webhooks;
+};
+
+const unformatHeaders = headers => {
```

Comment:
If you apply the following changes you won't be needing the `cleanHeaders` in the `cleanData` function.

```
const unformatHeaders = headers => {
  return headers.reduce((acc, current) => {
    const { key, value } = current;
    if (key !== '') {
      return {
        ...acc,
        [key]: value,
      };
    }

    return acc;
  }, {});
};
```

Reply:
And since you don't need anymore in your `checkFormErrors` you can completely delete it.